tools/google-cloud-support-slackbot/README.md

```diff
@@ -0,0 +1,153 @@
+	1. **Org Viewer**
+4. The **email addresses** of the people that will be enabling the API in the project
+
+Your Techincal Account Manager will file a request with the Support API team to give you access. The team typically processes these requests within 24 hours
+
+## Setup Part 2 - Google Cloud Phase 1 
+
+In the first phase of our Google Cloud setup, we will verify that our network is setup properly, create a lightweight VM to house our bot, and enable our Cloud Support API and create ourselves an API key. Go to [Google Cloud](https://cloud.google.com/console). **These steps need to be carried out in the project you specified in Part 1 of this setup guide.**
+
+### Networking
+
+From **VPC network > Firewall rules**, verify rules exist to **allow SSH and HTTP**.
+
+1. **If your project doesn't have a VPC, you will need to create one from VPC networks**. Select **Automatic** for your Subnet creation mode, and **allow-ssh** from **Firewall rules**
+2. If it doesn't exist, create the following firewall rule:
+	1. Name: `default-allow-http`
+	1. Priority: `1000`
+	1. Direction: `Ingress`
+	1. Action on match: `Allow`
+	1. Targets: `Specified target tags`
+	1. Target tags: `http-server`
+	1. Source filter: `IP ranges`
+	1. Source IP ranges: `0.0.0.0/0`
+	1. Protocols and Ports: `Specified protocols and ports`
+	1. tcp: `80`
+3. If an SSH firewall rule doesn't exist, create the following firewall rule:
+	1. Name: `default-allow-ssh`
+	1. Priority: `65534`
+	1. Direction: `Ingress`
+	1. Action on match: `Allow`
+	1. Targets: `All instances in the network`
+	1. Source filter: `IP ranges`
+	1. Source IP ranges: `0.0.0.0/0`
+	1. Protocols and Ports: `Specified protocols and ports`
+	1. tcp: `22`
+
+*Note that if you had to create the SSH firewall rule in Step 3, you will want to disable it after you complete the entire setup*
+
+### VM
+
+Go to **Compute Engine > VM instances** and perform the following:
+
+1. Click **+ Create Instance**
+	1. Under **Machine Configuration**, set the **Machine type** field to **e2-micro**. This should suffice for most implementations. If your team makes heavy use of the Cloud Support and the bot, you may need to upgrade the machine type
+	1. Under **Identity and API access > Service Account**, select your **service account** that was allow listed for the Cloud Support API
+	1. Under **Firewall**, select **Allow HTTP traffic**. If this option isn't available and you create the firewall rule in the Networking steps, then you will want to contact your Networking team about policies that may be preventing HTTP traffic
+	1. Click to expand **Management, security, disks, networking, sole tenancy**
+		1. Select the **Networking** tab
+		1. Under **Network interfaces**, click the network interface box
+			1. Set **Network** to the VPC where you have your firewall rules
+			1. Under **External IP**, select **Create IP address**. Choose whichever name and network service tier you prefer
+	1. Click **Create**
```

Comment:
All setup above can be performed in Terraform or Google Cloud Resource Manager. Is there any guidelines for that repository when Terraform should be used and when plain text tutorial? See example https://github.com/GoogleCloudPlatform/professional-services/tree/main/examples/kerberized_data_lake .

Reply:
I think the answer is it depends on each team's maturity with Google Cloud. For an organization that is new to cloud, following the step-by-step guide will likely make the most sense. For more experienced teams though, they will likely have terraform already handy that they can use to build up vanilla projects with most of this implemented